views/js/pciCreator/dev/mathEntryInteraction/runtime/mathEntryInteraction.js

```diff
@@ -36,14 +36,68 @@ define([
     'use strict';
 
     var ns = '.mathEntryInteraction';
+    var rootClass = 'mq-root-block';
+    var newLineClass = 'mq-tao-br';
+    var autoWrapClass = 'mq-tao-wrap';
+    var rootSelector = '.' + rootClass;
+    var newLineSelector = '.' + newLineClass;
+    var autoWrapSelector = '.' + autoWrapClass;
     var MQ = MathQuill.getInterface(2);
     var mathEntryInteraction;
 
+    /**
+     * Builds a simple HTML markup.
+     * @param {String} cls
+     * @param {String} [tag='div']
+     * @returns {String}
+     */
+    function htmlMarkup(cls, tag) {
+        tag = tag || 'div';
+        return '<' + tag + ' class="' + cls + '"></' + tag + '>';
+    }
+
+    /**
+     * Computes the full width of an element, plus its margin.
+     * This approach is more reliable than jQuery, as the decimals part is taken into account.
+     * @param element
+     * @returns {Number}
+     */
+    function getWidth(element) {
+        var style = element.currentStyle || window.getComputedStyle(element);
+        var rect = element.getBoundingClientRect();
+        var borderBox = style.boxSizing === 'border-box';
+        return rect.width + parseFloat(style.marginLeft) + parseFloat(style.marginRight) +
+            (borderBox ? 0 : parseFloat(style.paddingLeft) + parseFloat(style.paddingRight)) +
+            (borderBox ? 0 : parseFloat(style.borderLeftWidth) + parseFloat(style.borderRightWidth));
+    }
+
+    /**
+     * Auto wraps the content to avoid overflow
+     * @param $container
+     */
+    function applyAutoWrap($container) {
+        var maxWidth = $container.width();
+        var lineWidth = 0;
+        $container.find(autoWrapSelector).remove();
+        $container.children().each(function() {
+            var width = getWidth(this);
+            if (this.classList.contains(newLineSelector)) {
+                lineWidth = 0;
+            } else {
+                if (lineWidth + width >= maxWidth) {
+                    $(this).before(htmlMarkup(autoWrapClass));
+                    lineWidth = 0;
+                }
+                lineWidth += width;
+            }
+        });
+    }
+
```

Comment:
a nice improvement could be to keep track of the last space index (if any) and then insert the line break at this position. If no space is found then just keep current behavior.